src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -288,6 +288,45 @@ public void waitForAlertPresence() {
     }
 
     /**
+     * Waits for an alert modal to appear and dismisses it
+     */
+    public void waitForAndDismissAlertModal() {
+        waitForModalPresence();
+        WebElement okayButton = browser.driver.findElement(By.className("modal-btn-ok"));
+        waitForElementToBeClickable(okayButton);
+        okayButton.click();
+        waitForModalToDisappear();
+    }
+
+    /**
+     * Waits for a confirmation modal to appear and click the confirm button
+     */
+    public void waitForConfirmationModalAndClickOk() {
+        waitForAndDismissAlertModal();
```

Comment:
Let's put the content of `waitForAndDismissAlertModal` here, and let that function call this one instead. The "click OK to dismiss" relationship makes more sense compared to "dismiss to click OK".